dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -537,14 +542,10 @@ private void readDataInFetch()
         // Loop here while data is being consumed quickly enough
         while (remainingDataInCurrentRequest > 0)
         {
-            try
-            {
-                this.bufferWriteMutex.acquire();
-            }
-            catch(Exception e){}
+            this.bufferWriteMutex.acquireUninterruptibly();
```

Comment:
In this case there isn't a way for us to back out if the mutex has already been acquired. In order to continue parsing we have to acquire that mutex. However, this will not block forever unless there is a deadlock bug in the code. Which is always a potential issue, so I don't know what else we could do beyond checking to make sure we always release the lock appropriately.